py/py.mk

```diff
@@ -269,7 +269,10 @@ $(HEADER_BUILD)/compressed.data.h: $(HEADER_BUILD)/compressed.collected
 # build a list of registered modules for py/objmodule.c.
 $(HEADER_BUILD)/moduledefs.h: $(SRC_QSTR) $(QSTR_GLOBAL_DEPENDENCIES) | $(HEADER_BUILD)/mpversion.h
 	@$(ECHO) "GEN $@"
-	$(Q)$(PYTHON) $(PY_SRC)/makemoduledefs.py --vpath="., $(TOP), $(USER_C_MODULES)" $(SRC_QSTR) > $@
+	$(Q)echo $(TOP) > $(BUILD)/top.objs
+	$(Q)echo $(USER_C_MODULES) > $(BUILD)/user_c_modules.objs
+	$(Q)echo $(SRC_QSTR) > $(BUILD)/scr_qstr.objs
+	$(Q)$(PYTHON) $(PY_SRC)/makemoduledefs.py --spath="., @$(BUILD)/top.objs, @$(BUILD)/user_c_modules.objs" @$(BUILD)/scr_qstr.objs > $@
```

Comment:
`--spath` is not a valid option

also, does it make sense to use `$(ECHO)` instead of `$(Q)echo`?

Reply:
indeed `$(ECHO)` is more consistent 
and `--spath` was the problem ,both resolved